java/semantickernel-api/src/main/java/com/microsoft/semantickernel/plugin/KernelFunctionFactory.java

```diff
@@ -1,140 +1,60 @@
-
 import java.lang.reflect.Method;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
-import javax.annotation.Nullable;
 
 /**
  * Factory for creating {@link KernelFunction} instances.
  */
 public class KernelFunctionFactory {
 
     /**
-     * Creates a {@link KernelFunction} instance for a method, specified via a {@link Method} instance
-     * @param <T> The return type of the method.
+     * Creates a {@link KernelFunction} instance for a method, specified via a {@link Method}
+     * instance
+     *
+     * @param <T>    The return type of the method.
      * @param method The method to be represented via the created {@link KernelFunction}.
-     * @param target The target object for the {@code method} if it represents an instance method. This should be {@code null} if and only if {@code method} is a static method.
-     * @param functionName Optional function name. If {@code null}, it will default to one derived from the method represented by {@code method}.
-     * @param description Optional description of the method. If {@code null}, it will default to one derived from the method represented by {@code method}, 
-     *                    if possible (e.g. via a {@link com.microsoft.semantickernel.plugin.annotations.DefineKernelFunction} on the method).
-     * @param parameters Optional parameter descriptions. If {@code null}, it will default to one derived from the method represented by {@code method}.
-     * @param returnParameter Optional return parameter description. If {@code null}, it will default to one derived from the method represented by {@code method}.
+     * @param target The target object for the {@code method} if it represents an instance method.
+     *               This should be {@code null} if and only if {@code method} is a static method.
      * @return The created {@link KernelFunction} wrapper for {@code method}.
      */
-    public static <T> KernelFunction<T> createFromMethod(
-        Method method,
-        Object target,
-        @Nullable String functionName,
-        @Nullable String description,
-        @Nullable List<KernelParameterMetadata<?>> parameters,
-        @Nullable KernelReturnParameterMetadata<?> returnParameter) {
-        return KernelFunctionFromMethod.create(method, target, functionName, description,
-            parameters, returnParameter);
-    }
 
-
-    /**
-     * Creates a {@link KernelFunction} instance for a prompt specified via a prompt template.
-     *
-     * @param promptTemplate Prompt template for the function.
-     * @param <T>           Type of the value that the function will return
-     * @return The created {@link KernelFunction} for invoking the prompt.
-     */
-    public static <T> KernelFunction<T> createFromPrompt(String promptTemplate) {
-        return createFromPrompt(promptTemplate, null, null, null, null, null);
+    public static <T> KernelFunctionFromMethod.Builder<T> createFromMethod(
```

Comment:
Can all of this be moved to KernelFunction and this factory class gotten rid of?